test/framework/pod.go

```diff
@@ -42,3 +42,20 @@ func (f *Framework) PrintPodLogs(ns, p string) error {
 
 	return nil
 }
+
+// GetRestartCount returns a map of container names and their restart counts for
+// a given pod.
+func (f *Framework) GetRestartCount(ns, podName string) (map[string]int32, error) {
+	restarts := map[string]int32{}
+
+	pod, err := f.KubeClient.CoreV1().Pods(ns).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return map[string]int32{}, errors.Wrap(err, "failed to retrieve pod to get restart count")
```

Comment:
you could return `restarts` since you already instantiated one empty map. or better yet, just return `nil, err`. In Go, slices and maps are both pointer types so you can return nil in their place. If this function returns an error, there is no guarantee that the map should be iterable.